bgpy/simulation_engine/policies/aspa/aspa.py

```diff
@@ -27,9 +27,10 @@ def _valid_ann(self, ann: "Ann", from_rel: Relationships) -> bool:  # type: igno
 
         assert len(set(ann.as_path)) == len(ann.as_path), "We deal with prepending"
 
+        # Note: This first if check has to be removed if you want to implement
+        # route server to RS-Client behaviour
         if ann.next_hop_asn != ann.as_path[0]:
             return False
```

Comment:
You make a great point here, do you think it may be possible to move this next_hop check into it's own function? That way if a user wanted to implement route servers, all they'd have to do is subclass ASPA and modify that one hook function rather than modifying the entire valid_ann func? Great catch I didn't think of that!